src/OpenLoco/Utility/Numeric.cpp

```diff
@@ -39,14 +45,20 @@ namespace OpenLoco::Utility
         uint8_t success = _BitScanReverse(&i, source);
         return success != 0 ? i : -1;
 #elif defined(__GNUC__)
-        int32_t success = __builtin_ffs(__builtin_bswap32(source));
-        return success - 1;
+        int32_t success = source == 0 ? -1 : __builtin_clz(source) ^ 31;
```

Comment:
i had a thought that success is the incorrect name for this the only reason we have used that name is because of the Windows function uses a success return value but these intrinsics are really returning just the result.